sshutil/sshutil.go

```diff
@@ -1,59 +1,113 @@
 package sshutil
 
 import (
-	"bytes"
-	"crypto/rand"
-	"crypto/rsa"
-	"crypto/x509"
+	"math/rand"
+	"crypto/ed25519"
 	"encoding/pem"
+	"bytes"
 
 	"golang.org/x/crypto/ssh"
 )
 
+// SSHKey stores byte slices that represent private and public ssh keys
 type SSHKey struct {
 	Private []byte
 	Public  []byte
 }
 
+// NewPrivatePublicKeyPair generates a an ed25519 ssh private and public key
 func NewPrivatePublicKeyPair(bits int) (SSHKey, error) {
 	var keys SSHKey
 
-	privateKey, err := NewPrivateKey(bits)
+	pub, priv, err := ed25519.GenerateKey(nil)
 	if err != nil {
-		return keys, err
+		return SSHKey{}, err
 	}
 
-	keys.Public, err = NewPublicKey(privateKey)
+	keys.Public, err = newPublicKey(pub)
 	if err != nil {
-		return keys, err
+		return SSHKey{}, err
 	}
 
-	keys.Private, err = RSAToPEM(privateKey)
+	keys.Private, err = newPrivateKey(priv)
 	if err != nil {
-		return keys, err
+		return SSHKey{}, err
 	}
 
 	return keys, nil
 }
 
-func NewPrivateKey(bits int) (*rsa.PrivateKey, error) {
-	pk, err := rsa.GenerateKey(rand.Reader, bits)
+func newPublicKey(key ed25519.PublicKey) ([]byte, error) {
+	pubKey, err := ssh.NewPublicKey(key)
 	if err != nil {
-		return pk, err
+		return nil, err
 	}
+	return ssh.MarshalAuthorizedKey(pubKey), nil
+}
 
-	err = pk.Validate()
-	if err != nil {
-		return pk, err
+func newPrivateKey(key ed25519.PrivateKey) ([]byte, error) {
+	// https://anongit.mindrot.org/openssh.git/tree/PROTOCOL.key?h=V_8_1_P1
```

Comment:
What's the purpose of this comment? Why are we linking to this? Do we need to link to this?

Please provide more context around this code.

Reply:
This link was in the piece of code that chris shared to generate the keys. I found it helpful in understanding why the scripts were setup that way so i left it in. It gives some context to why the structs are setup the way there are. I can be removed but if that context is needed it would need to be added as other comments

Reply:
Great. Please explain that in the code in terms of what it explains, and then use the URL as your reference 😉